src/EventSubscriber/SmsMessageProcessor.php

```diff
@@ -295,6 +332,8 @@ public static function getSubscribedEvents() {
     $events[SmsEvents::MESSAGE_PRE_PROCESS][] = ['chunkMaxRecipients', -1024];
     // Ensure reports for outgoing messages.
     $events[SmsEvents::MESSAGE_OUTGOING_POST_PROCESS][] = ['ensureReportsPostprocess', 1024];
+    // Update delivery reports as they are received.
+    $events[SmsEvents::DELIVERY_REPORT_POST_PROCESS][] = ['updateDeliveryReports', 1024];
```

Comment:
`::updateDeliveryReports` belongs in its own class, since \Drupal\sms\EventSubscriber\SmsMessageProcessor is dedicated to messages.